internal/app/dashboard/main.go

```diff
@@ -0,0 +1,54 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+// Package dashboard implements dashboard functionality.
+package dashboard
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/metadata"
+
+	"github.com/siderolabs/talos/internal/pkg/dashboard"
+	"github.com/siderolabs/talos/pkg/grpc/middleware/authz"
+	"github.com/siderolabs/talos/pkg/machinery/client"
+	"github.com/siderolabs/talos/pkg/machinery/constants"
+	"github.com/siderolabs/talos/pkg/machinery/role"
+	"github.com/siderolabs/talos/pkg/startup"
+)
+
+// Main is the entrypoint into dashboard.
+func Main() {
+	if err := dashboardMain(); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func dashboardMain() error {
+	ctx, cancel := signal.NotifyContext(context.Background(), syscall.SIGTERM, syscall.SIGINT)
```

Comment:
I think we should have a way to disable `SIGINT` behavior when running on a tty - same as `q` button. There should be no way to quit the dashboard :)

Reply:
done, no way to exit now :D